publisher-sdk/src/main/java/com/criteo/publisher/privacy/ConsentData.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.privacy
+
+import com.criteo.publisher.annotation.OpenForTesting
+import java.util.concurrent.atomic.AtomicReference
+
+@OpenForTesting
+class ConsentData {
+
+  /**
+   * Consent can be in any of the following three states:
+   *
+   * <ul>
+   *  <li> CONSENT_GIVEN: it is known that consent has been given by the user</li>
+   *  <li> CONSENT_NOT_GIVEN: it is known that consent has not been given by the user,
+   *  or isn't known yet.</li>
+   * </ul>
+   */
+  enum class ConsentStatus {
```

Comment:
Do you have a reason to use an enum value rather than a simple boolean ?
Using enum makes implementation a little bit harder to get.

Reply:
I feel it is cleaner and makes the code easier to read. It's unclear seeing `true/false` being passed around vs. seeing values such as these ones.

Reply:
It might be cleaner internally to the `ConsentData` class. Although, even if there was a third state, I would argue that callers should not be able to know about internal state and should only consume a boolean decision: do I have the consent or not ?

Reply:
Yeah I will move it to boolean in another PR.